packages/shared/src/components/drawers/NavDrawer.tsx

```diff
@@ -64,7 +64,16 @@ export function NavDrawer({
       )}
       <NavDrawerContent>
         {items.map((item) => (
-          <NavDrawerItem key={item.label} {...item} />
+          <NavDrawerItem
+            key={item.label}
+            {...item}
+            onClick={(e) => {
+              if (item.onClick) {
+                item.onClick(e);
+              }
+              ref.current?.onClose();
+            }}
+          />
```

Comment:
Why is this needed?
Can't we uniform it inside the NavDrawerItem maybe?

Reply:
The ref is not found in the item itself. Unless we create a context or something to access this close function, we will have to make the call in this function.

Reply:
Hmm couldn't it just accept a ref as prop?
(anyway not breaking, more of a thought)

Reply:
Yeah, we can do that. Clicking a nav item should automatically render a close action.